karate-core/src/main/java/com/intuit/karate/driver/Driver.java

```diff
@@ -90,6 +90,8 @@
 
     void setDimensions(Map<String, Object> map); // setter
 
+    void setEmulateDevice(Map<String, Object> emulateDeviceParams);
```

Comment:
this does not need to be on the Driver interface, can be left on DevToolsDriver - it will still work in feature files, but do confirm

Reply:
@ptrthomas  Yes, this works on feature files without having to configure at driver level. Couple of reasons why i tried to configure at the driver level is 

1.  My initial page has to be some url which i felt wouldn't be a good experience.
2. I see that the chromedriver and other drivers can emulate this behavior only if set in the options when starting the browser and cannot be modified once the browser has started.  (though i could never get it working by passing it as options when chromewebdriver/geckowebdriver/others)

Having said that, I see that 1) can be solved by visiting blank url as you suggested below and that works.

**Given** driver 'about:blank'
**And** driver.emulateDevice = {width: 375, height: 812, userAgent: 'Mozilla/5.0 (iPhone; CPU iPhone OS 13_2_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Mobile/15E148 Safari/604.1'}
**Given** driver 'https://www.intuit.com'

Please suggest if this still should not be a configurable property at driver level and I will make the necessary changes, and also update the readme file.

Reply:
> though i could never get it working by passing it as options when chromewebdriver/geckowebdriver/other

if you get it working, then that's a different story. but now I prefer the "chrome-only" option